File: CS 185 - Computing in Developing Regions/ListenUp-Node/make_call.js
require('dotenv').config();
const twilio = require('twilio');

function makeCall(to, zip){
    const accountSid   = process.env.ACCOUNT_SID;
    const authToken    = process.env.AUTH_TOKEN;
    const server       = process.env.SERVER;
    const client = twilio(accountSid, authToken);

    action = `${server}/choose_option`
    instructions = ` \
    <Response> \
        <Say voice="alice">Thank you for calling Global News.</Say> \
        <Gather action="${action}" numDigits="3" timeout="10"> \
            <Say voice="alice">Please select 1 for trending local news, 2 for trending world news, 3 if you have a specific query, or 4 to repeat these options.</Say> \
        </Gather> \
    </Response> \
    `
    // Make the initial outgoing call
    client.calls.create({
    //   url: server+'/choose_option',
      to: to,
      from: "+18449504572",
      twiml: instructions,
    })
    .then(call => console.log(call.sid))
    .catch(error => console.log(error));
}

module.exports = { makeCall };

File: CS 185 - Computing in Developing Regions/ListenUp/make_call.py
# Download the helper library from https://www.twilio.com/docs/python/install
from twilio.rest import Client
from dotenv import load_dotenv
import os


def make_call(to, zip):

    # Find your Account SID and Auth Token at twilio.com/console
    # and set the environment variables
    load_dotenv()
    account_sid = os.environ['ACCOUNT_SID']
    auth_token = os.environ['AUTH_TOKEN']
    server = os.environ['SERVER']
    
    client = Client(account_sid, auth_token)

    action = f"{server}/choose_option"
    instructions = f''' 
    <Response>
        <Say voice="alice">Thank you for calling Global News.</Say>
        <Redirect method="GET">
            {server}/choose_option?Digits=Begin
        </Redirect>
    </Response>
    '''

    call = client.calls.create(
        twiml=instructions,
        to=to,
        from_='+18449504572'
    )

    print(call.sid)
